libs/shared/contentful/src/lib/contentful.client.ts

```diff
@@ -50,6 +60,40 @@ export class ContentfulClient {
     }
   }
 
+  private async getLocales(): Promise<string[]> {
+    if (!!this.locales?.length) {
+      return this.locales;
+    }
+
+    try {
+      const localesUrl = `${this.contentfulClientConfig.cdnApiUri}/spaces/${this.contentfulClientConfig.graphqlSpaceId}/environments/${this.contentfulClientConfig.graphqlEnvironment}/locales?access_token=${this.contentfulClientConfig.graphqlApiKey}`;
+      const response = await fetch(localesUrl);
+      const results = await response.json();
```

Comment:
I believe you'll need to move this down below the `response.ok` check since a failed response may not have a consumable JSON body. I may be wrong though!